core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -292,9 +295,18 @@ protected Optional<Expression> translateCall(io.trino.spi.expression.Call call)
                 return translateInPredicate(call.getArguments().get(0), call.getArguments().get(1));
             }
 
-            ResolvedFunction resolved = plannerContext.getMetadata().resolveBuiltinFunction(
-                    call.getFunctionName().getName(),
-                    fromTypes(call.getArguments().stream().map(ConnectorExpression::getType).collect(toImmutableList())));
+            ResolvedFunction resolved;
+            if (JSON_STRING_TO_MAP_NAME.equals(call.getFunctionName().getName()) ||
+                    JSON_STRING_TO_ARRAY_NAME.equals(call.getFunctionName().getName()) ||
+                    JSON_STRING_TO_ROW_NAME.equals(call.getFunctionName().getName())) {
+                // These are special functions that currently need to be resolved via getCoercion() -- TODO: fix this
+                resolved = plannerContext.getMetadata().getCoercion(builtinFunctionName(call.getFunctionName().getName()), call.getArguments().get(0).getType(), call.getType());
```

Comment:
Isn't `getCoercion` API is a bit tricky - The internal logic which we do to resolving the function - but the API looks like we are applying coercion from one type to another